app/controllers/orders_controller.rb

```diff
@@ -1,10 +1,10 @@
-class OrdersController < ApplicationController
+  class OrdersController < ApplicationController
   before_action :set_order, only: [:show, :edit, :update, :destroy]
-
+  before_action :authenticate_dispatcher!
```

Comment:
Only for edit and update. By the way, we shouldn't destroy orders, only cancel them.